### PR TITLE
[nextra] Migrate Move docs to Nextra

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
1. Cleans up and ensures all code samples are valid Move
2. Fixes all code blocks to have highlighting properly
3. Fixes some logical issues in code
4. No updates to word content

### Checklist

- Do all Lints pass?
  - [x] Have you ran `pnpm spellcheck`? (there are a ton of errors here, we need to clean up separately)
  - [x] Have you ran `pnpm fmt`?
  - [x] Have you ran `pnpm lint`?

### CODE REVIEW

**Comment by hariria** [comment]
Thanks for doing this. When looking at the preview for move, the page doesn't seem to be loading. Can you check and see if you set the `_meta.tsx` correctly for it? (https://developer-docs-nextra-git-greg-migrate-nextra-aptoslabs.vercel.app/en/build/move)

Also I was under the impression we were going to keep it as `Smart Contracts` based on discussion with Zihan since first-timers may not know what Move is

File: apps/nextra/pages/en/build/move/book/references.mdx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+    read_and_assign(y, x); // valid
+    read_and_assign(x, y); // invalid!
+  }
+}
+```
+
+will yield the following error messages
+
+```bash
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/book/references.mdx:182]: Nit: for all the bash codeblocks can we cmd + shift + f and replace to

```bash filename="Terminal"

File: apps/nextra/pages/en/build/move/book/SUMMARY.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# The Move Programming Language
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/book/SUMMARY.mdx:1]: I think the files in the book directory are missing frontmatter for the title of the page. e.g.,

```
---
title: "Summary"
---
```

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+   This results in the following file structure:
+
+   ```text
+   testing/
+      Move.toml
+      sources/
+         my_script.move
+         my_module.move
+   ```
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx:94]: Can we use `<Filetree />` here (https://preview.aptos.dev/en/docs/components/filetree#usage)

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+   INCLUDING DEPENDENCY AptosFramework
+   INCLUDING DEPENDENCY AptosStdlib
+   INCLUDING DEPENDENCY MoveStdlib
+   BUILDING run_script
+   {
+     "Result": [
+       "cb265645385819f3dbe71aac266e319e7f77aed252cacf2930b68102828bf615::my_module"
+     ]
+   }
+   ```
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx:110]: Nit: change to ```bash filename="Terminal

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+   Note how we use the `--named-addresses` argument. This is necessary because in the code we refer to this named address called `addr`. The compiler needs to know what this refers to. Instead of using this CLI argument, you could put something like this in your `Move.toml`:
+
+   ```toml
+   [addresses]
+   addr = "cb265645385819f3dbe71aac266e319e7f77aed252cacf2930b68102828bf615"
+   ```
+
+6. Run the compiled script:
+   ```sh
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx:120]: Nit: change to ```bash filename="Terminal

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+See the [code](https://github.com/banool/move-examples/tree/main/run_script) used for this document. The full example explains how to use a Move script that relies on a user-created Move module as well.
+
+See also how to do this with the [Rust SDK](https://github.com/aptos-labs/aptos-developer-discussions/discussions/24) instead of the Aptos CLI in Aptos Developer Discussions.
+
+## Advanced
+
+You may execute a script in a more streamlined fashion; instead of running `aptos move compile` and then `aptos move run-script --compiled-script-path` separately, you can just do this:
+
+```sh
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx:149]: Nit: change to ```bash filename="Terminal

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/compiling-scripts.mdx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+# How can I compile Move Scripts?
+
+Move scripts can be compiled with the already existing Aptos Move compiler in
+the Aptos CLI. For more on how to install and use the Aptos CLI with Move contracts, go to the [Working With Move Contracts](../../../tools/aptos-cli/use-cli/working-with-move-contracts.md) page.
+
+Once you have the Aptos CLI installed, you can compile a script by running the following command from within the script package:
+
+```shell
+aptos move compile
+```
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/compiling-scripts.mdx:14]: Nit: change to ```bash filename="Terminal

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/compiling-scripts.mdx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+INCLUDING DEPENDENCY AptosStdlib
+INCLUDING DEPENDENCY MoveStdlib
+BUILDING transfer_half
+{
+  "Result": {
+    "script_location": "/opt/git/developer-docs/apps/docusaurus/static/move-examples/scripts/transfer_half/script.mv",
+    "script_hash": "9b57ffa952da2a35438e2cf7e941ef2120bb6c2e4674d4fcefb51d5e8431a148"
+  }
+}
+```
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/compiling-scripts.mdx:63]: Nit: change to ```bash filename="Terminal

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/running-scripts.mdx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+```shell
+aptos move run-script --compiled-script-path /opt/git/developer-docs/apps/docusaurus/static/move-examples/scripts/transfer_half/script.mv
+```
+
+Similarly, if it's not compiled, just use `--script-path`
+
+```shell
+aptos move run-script --script-path ./sources/transfer_half.move
+```
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/running-scripts.mdx:105]: Nit: change to ```bash filename="Terminal

File: apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+       account: cb265645385819f3dbe71aac266e319e7f77aed252cacf2930b68102828bf615
+       rest_url: "https://api.devnet.aptoslabs.com"
+       faucet_url: "https://faucet.devnet.aptoslabs.com"
+   ```
+
+3. From this same directory, initialize a new Move project:
+
+   ```sh
+   aptos move init --name run_script
+   ```
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx:76]: Nit: change to ```bash filename="Terminal

**Review by hariria — COMMENTED** [review]
Thanks for doing this. I haven't had a chance to take a look at the preview yet but here's some things I would change:

1. I was under the impression the section was `Smart Contract`, not `Move`
2. You changed the filenames to `.mdx` which is great. We should also change the file references to be `filename.md` to `filename.mdx`
3. Missing title frontmatter for some content (see comments)
4. ```sh or ```bash or ```shell -> ```bash filename="Terminal". Hopefully should be pretty easy to do a findall and replace
5. Filetree component (see comments)

Will do another review once I can see the preview

**Comment by gregnazario** [comment]
> Thanks for doing this. When looking at the preview for move, the page doesn't seem to be loading. Can you check and see if you set the `_meta.tsx` correctly for it? (https://developer-docs-nextra-git-greg-migrate-nextra-aptoslabs.vercel.app/en/build/move)
> 
> Also I was under the impression we were going to keep it as `Smart Contracts` based on discussion with Zihan since first-timers may not know what Move is

I copied it over completely, and didn't want to modify it too much, we can rename and move after the migration.

File: apps/nextra/pages/en/build/move/book/SUMMARY.mdx (hunk 2 above)
Comment (gregnazario) [reply apps/nextra/pages/en/build/move/book/SUMMARY.mdx:1]: The frontmatter isn't required it seems to fill them out automatically

File: apps/nextra/pages/en/build/move/_meta.tsx [hunk 11]
```diff
@@ -0,0 +1,18 @@
+export default {
+  index: {
+    title: "Introduction",
+  },
+  book: {
+    title: "Move Book",
+    theme: {
+      toc: false,
+      layout: "full",
```
Comment (hariria) [inline apps/nextra/pages/en/build/move/_meta.tsx:9]: Can we remove `layout: "full"` and `toc: false` here? Unfortunately when you do that you lose  the breadcrumb at the top of each page ([example](https://developer-docs-nextra-git-greg-migrate-nextra-aptoslabs.vercel.app/en/build/move/book/SUMMARY)). 

I looked at each page in the Move book, I think the only ones that would benefit from full layout are the ones with wider columns. I've noted each one below. 

Places to apply layout full
1. Vector -> https://developer-docs-nextra-git-greg-migrate-nextra-aptoslabs.vercel.app/en/build/move/book/vector
2. References -> https://developer-docs-nextra-git-greg-migrate-nextra-aptoslabs.vercel.app/en/build/move/book/references
3. Tuples and Unit -> https://developer-docs-nextra-git-greg-migrate-nextra-aptoslabs.vercel.app/en/build/move/book/tuples

File: apps/nextra/pages/en/build/move/book/SUMMARY.mdx (hunk 2 above)
Comment (hariria) [reply apps/nextra/pages/en/build/move/book/SUMMARY.mdx:1]: Yeah I think it uses the sidebar titles as a fallback if the frontmatter title isn't filled in

**Review by hariria — APPROVED** [review]

**Pull request merged**